src/pal/src/thread/process.cpp

```diff
@@ -2718,13 +2704,20 @@ DestroyProcessModules(IN ProcessModules *listHead)
 __attribute__((destructor)) 
 void PROCNotifyProcessShutdown()
 {
-    TRACE("PROCNotifyProcessShutdown %p\n", g_shutdownCallback.RawValue());
-
+    // Call back into the coreclr to clean up the debugger transport pipes
     PSHUTDOWN_CALLBACK callback = InterlockedExchangePointer(&g_shutdownCallback, NULL);
     if (callback != NULL)
     {
         callback();
     }
+
+    // Cleanup the name continue semaphore on exit and abormal terminatation
+    sem_t *continueSem = InterlockedExchangePointer(&g_continueSem, SEM_FAILED);
+    if (continueSem != SEM_FAILED)
+    {
+        sem_close(continueSem);
+        sem_unlink(g_continueSemName);
```

Comment:
If we are using the old name, should it be the debugger's responsibility to do this unlink?